extern-api-java-sdk/src/main/java/ru/kontur/extern_api/sdk/service/impl/TaskServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package ru.kontur.extern_api.sdk.service.impl;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import ru.kontur.extern_api.sdk.httpclient.api.DraftsApi;
+import ru.kontur.extern_api.sdk.model.*;
+import ru.kontur.extern_api.sdk.provider.AccountProvider;
+import ru.kontur.extern_api.sdk.service.TaskService;
+import ru.kontur.extern_api.sdk.utils.ConditionAwaiter;
+
+public class TaskServiceImpl implements TaskService {
+
+    private final AccountProvider acc;
+    private final DraftsApi api;
+    private final static int DELAY_TIMEOUT = 2000;
+    private final UUID draftId;
+
+    TaskServiceImpl(AccountProvider accountProvider, DraftsApi api, UUID draftId) {
+        this.acc = accountProvider;
+        this.api = api;
+        this.draftId = draftId;
+    }
+
+    @Override
+    public CompletableFuture<SendTaskInfo> startSendAsync() {
+        return api.startSend(acc.accountId(), draftId, false);
+    }
+
+    @Override
+    public CompletableFuture<CheckTaskInfo> startCheckAsync() {
+        return api.startCheck(acc.accountId(), draftId)
+                .thenApply(WrappedCheckDataTaskInfo::unwrap);
+    }
+
+    @Override
+    public CompletableFuture<PrepareTaskInfo> startPrepareAsync() {
+        return api.startPrepare(acc.accountId(), draftId);
+    }
+
+
+    @Override
```

Comment:
Немного о код стайле
```suggestion
    @Override
    public CompletableFuture<CheckResultData> getCheckResult(CheckTaskInfo checkTaskInfo) {
        return waitWhileRunning(() -> api
                .getCheckResult(acc.accountId(), draftId, checkTaskInfo.getId())
                .thenApply(WrappedCheckDataTaskInfo::unwrap)
        );
    }
```

Reply:
done